Fix num_iters > 5 Shiftpointer issue

billmguo opened the pull request:
Summary: in the original implementation when num_iters > 5 it will crash during exit

Differential Revision: D70974749

cccclai commented:
Can you explain what is the issue?

billmguo commented:
@pytorchbot label "topic: not user facing"

billmguo commented:
> 

03-11 09:52:35.461 21905 21905 F DEBUG   :       #00 pc 000000000005b730  /apex/com.android.runtime/lib64/bionic/libc.so (abort+168) (BuildId: edc1b601d9f8344c1eccd3d77a458f65)
03-11 09:52:35.461 21905 21905 F DEBUG   :       #01 pc 0000000000048f20  /apex/com.android.runtime/lib64/bionic/libc.so (scudo::die()+12) (BuildId: edc1b601d9f8344c1eccd3d77a458f65)
03-11 09:52:35.461 21905 21905 F DEBUG   :       #02 pc 00000000000498a8  /apex/com.android.runtime/lib64/bionic/libc.so (scudo::ScopedErrorReport::~ScopedErrorReport()+36) (BuildId: edc1b601d9f8344c1eccd3d77a458f65)
03-11 09:52:35.461 21905 21905 F DEBUG   :       #03 pc 0000000000049b08  /apex/com.android.runtime/lib64/bionic/libc.so (scudo::reportHeaderCorruption(void*)+100) (BuildId: edc1b601d9f8344c1eccd3d77a458f65)
03-11 09:52:35.461 21905 21905 F DEBUG   :       #04 pc 000000000004b82c  /apex/com.android.runtime/lib64/bionic/libc.so (scudo::Allocator<scudo::AndroidConfig, &(scudo_malloc_postinit)>::deallocate(void*, scudo::Chunk::Origin, unsigned long, unsigned long)+304) (BuildId: edc1b601d9f8344c1eccd3d77a458f65)
03-11 09:52:35.462 21905 21905 F DEBUG   :       #05 pc 000000000027d98c  /data/local/tmp/qnn_test_runner
03-11 09:52:35.462 21905 21905 F DEBUG   :       #06 pc 00000000000d20e4  /data/local/tmp/qnn_test_runner
03-11 09:52:35.462 21905 21905 F DEBUG   :       #07 pc 00000000000d2094  /data/local/tmp/qnn_test_runner
03-11 09:52:35.462 21905 21905 F DEBUG   :       #08 pc 00000000000d2038  /data/local/tmp/qnn_test_runner
03-11 09:52:35.462 21905 21905 F DEBUG   :       #09 pc 0000000000108da4  /data/local/tmp/qnn_test_runner
03-11 09:52:35.462 21905 21905 F DEBUG   :       #10 pc 0000000000108c24  /data/local/tmp/qnn_test_runner
03-11 09:52:35.462 21905 21905 F DEBUG   :       #11 pc 0000000000108bb4  /data/local/tmp/qnn_test_runner
03-11 09:52:35.463 21905 21905 F DEBUG   :       #12 pc 0000000000108a24  /data/local/tmp/qnn_test_runner
03-11 09:52:35.463 21905 21905 F DEBUG   :       #13 pc 0000000000108990  /data/local/tmp/qnn_test_runner
03-11 09:52:35.463 21905 21905 F DEBUG   :       #14 pc 000000000010894c  /data/local/tmp/qnn_test_runner
03-11 09:52:35.463 21905 21905 F DEBUG   :       #15 pc 00000000001088f8  /data/local/tmp/qnn_test_runner
03-11 09:52:35.463 21905 21905 F DEBUG   :       #16 pc 0000000000118870  /data/local/tmp/qnn_test_runner
03-11 09:52:35.464 21905 21905 F DEBUG   :       #17 pc 00000000000f6338  /data/local/tmp/qnn_test_runner
03-11 09:52:35.464 21905 21905 F DEBUG   :       #18 pc 00000000000f6398  /data/local/tmp/qnn_test_runner
03-11 09:52:35.464 21905 21905 F DEBUG   :       #19 pc 00000000001035e4  /data/local/tmp/qnn_test_runner
03-11 09:52:35.464 21905 21905 F DEBUG   :       #20 pc 000000000010360c  /data/local/tmp/qnn_test_runner
03-11 09:52:35.464 21905 21905 F DEBUG   :       #21 pc 00000000000d3bd0  /data/local/tmp/qnn_test_runner
03-11 09:52:35.464 21905 21905 F DEBUG   :       #22 pc 00000000000d3b44  /data/local/tmp/qnn_test_runner
03-11 09:52:35.464 21905 21905 F DEBUG   :       #23 pc 00000000000d39fc  /data/local/tmp/qnn_test_runner
03-11 09:52:35.464 21905 21905 F DEBUG   :       #24 pc 00000000000d0a38  /data/local/tmp/qnn_test_runner
03-11 09:52:35.464 21905 21905 F DEBUG   :       #25 pc 00000000000d0258  /data/local/tmp/qnn_test_runner
03-11 09:52:35.465 21905 21905 F DEBUG   :       #26 pc 0000000000053e50  /apex/com.android.runtime/lib64/bionic/libc.so (__libc_init+108) (BuildId: edc1b601d9f8344c1eccd3d77a458f65)

haowhsu-quic reviewed (approved):
Thank you.

cccclai reviewed (approved):
Approved on behalf of QC's team review

billmguo committed:
Fix num_iters > 5 Shiftpointer issue (#9150)

Summary:

in the original implementation when num_iters > 5 it will crash during exit

Reviewed By: cccclai

Differential Revision: D70974749
